File: ToolBag/ParseTool/src/parsetool/workflow/models/statements/ForStatement.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package parsetool.workflow.models.statements;

import parsetool.models.common.TokenPair;
import parsetool.workflow.models.State;
import parsetool.workflow.models.expressions.Expression;
import parsetool.workflow.models.expressions.Logical_Or_Expression;

/**
 *
 * @author Administrator
 */
public class ForStatement extends TokenPair {
    private Declare_Statement declare_statement;
    private Logical_Or_Expression logical_or_expression;
    private Expression expression;
    private State state;

    /**
     * @return the declare_statement
     */
    public Declare_Statement getDeclare_statement() {
        return declare_statement;
    }

    /**
     * @param declare_statement the declare_statement to set
     */
    public void setDeclare_statement(Declare_Statement declare_statement) {
        this.declare_statement = declare_statement;
    }

    /**
     * @return the logical_or_expression
     */
    public Logical_Or_Expression getLogical_or_expression() {
        return logical_or_expression;
    }

    /**
     * @param logical_or_expression the logical_or_expression to set
     */
    public void setLogical_or_expression(Logical_Or_Expression logical_or_expression) {
        this.logical_or_expression = logical_or_expression;
    }

    /**
     * @return the expression
     */
    public Expression getExpression() {
        return expression;
    }

    /**
     * @param expression the expression to set
     */
    public void setExpression(Expression expression) {
        this.expression = expression;
    }

    /**
     * @return the state
     */
    public State getState() {
        return state;
    }

    /**
     * @param state the state to set
     */
    public void setState(State state) {
        this.state = state;
    }
}

File: CodeHelper/CodeHelper.Core/Parse/ParseResults/Workflows/Statements/ForStatement.cs
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

using CodeHelper.Core.Parse.ParseResults;
using CodeHelper.Core.Parse.ParseResults.Workflows.Statements;
using CodeHelper.Core.Parse.ParseResults.Workflows.Expressions;
using CodeHelper.Core.Parse.ParseResults.Workflows;
namespace CodeHelper.Core.Parse.ParseResults.Workflows.Statements
{

    /**
     *
     * @author Administrator
     */
    public class ForStatement : TokenPair
    {
        public Declare_Statement Declare_statement { get; set; }
        public Logical_Or_Expression Logical_or_expression { get; set; }
        public Expression Expression { get; set; }
        public State State { get; set; }

        public ForStatementInfo Parse()
        {
            var rslt = new ForStatementInfo(this);
            rslt.Declare_statement = this.Declare_statement.Parse();
            rslt.Logical_or_expression = this.Logical_or_expression.Parse();
            rslt.Expression = this.Expression.Parse();
            rslt.Statements = this.State.Parse(null);
            return rslt;
        }
    }
}